test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresUnreferencedCodeCapability.cs

```diff
@@ -234,6 +237,60 @@ static void TestStaticCctorRequiresUnreferencedCode ()
 			_ = new StaticCtor ();
 		}
 
+		class StaticCtorTriggeredByFieldAccess
+		{
+			[RequiresUnreferencedCode ("Message for --StaticCtorTriggeredByFieldAccess.Cctor--")]
+			static StaticCtorTriggeredByFieldAccess ()
+			{
+				field = 0;
+			}
+
+			public static int field;
+		}
+
+		[ExpectedWarning ("IL2026", "--StaticCtorTriggeredByFieldAccess.Cctor--")]
+		static void TestStaticCtorMarkingIsTriggeredByFieldAccess ()
+		{
+			var x = StaticCtorTriggeredByFieldAccess.field + 1;
+		}
+
+		class TypeIsBeforeFieldInit
+		{
+			public static int field = AnnotatedMethod ();
+
+			[RequiresUnreferencedCode ("Message from --TypeIsBeforeFieldInit.AnnotatedMethod--")]
+			public static int AnnotatedMethod () => 42;
+		}
+
+		[LogContains ("IL2026: Mono.Linker.Tests.Cases.RequiresCapability.RequiresUnreferencedCodeCapability.TypeIsBeforeFieldInit..cctor():" +
```

Comment:
Why do we use LogContains here?

Reply:
As @sbomer puts above, Roslyn generates a .cctor and annotates the type with `beforefieldinit`, this static constructor initializes the field and thus calls `AnnotatedMethod`, triggering the warning from within the .cctor. Because of this, there's no place where we could put the `ExpectedWarning` attribute (the call is made from generated code).